### DISCUSSION TITLE
As a user, I want to upload an Excel file so that I don't have to manually input data

### DISCUSSION THREAD

**Discussion by GalenChang** [visualizing-your-future/visualizing-your-future#16] [General]
0) Have defined Excel sheet format(s) based on the type of data the user will be uploading.  This is so the application can parse the file(s) correctly.

1) Log in as user.
2) Navigate to "Data Input" page.
3) User selects a defined Excel sheet format.
4) Use upload feature to upload Excel file (instead of manually inputting data).
5) Application should parse data and store it in a database (temporarily?  permanently?).
6) Either return success or failure (and indicate why if possible, such as "X column is not part of recognized format Y").
7) Display the data in an organized manner (a simple table is fine.  This is for the user to verify the data was uploaded correctly).

**Comment by GalenChang** [comment]
Broke down the following epic into a smaller, more defined user story.

"As a member of the engagement team for Spire HI LLP, I want to convert complex Excel data into a dynamic visual format, so that I can present visually compelling financial insights."

Future/additional user stories can build off of this, such as converting the uploaded data to specific graphic representations (charts, etc).

**Comment by bri111** [comment]
As a user, I want to login with my credentials, so that I can securely access the dashboard and my personal data.

**Comment by GalenChang** [comment]
Feedback from professor on Wed (9/4)

This needs to be broken down more.  Some possible user stories include:

1) Create Excel sheet formats the app will parse.
2) Find methods to parse Excel sheet(s).
3) Properly store data in a database (MongoDB?).
